cnd/src/main.rs

```diff
@@ -52,6 +53,8 @@ fn main() -> anyhow::Result<()> {
 
     crate::trace::init_tracing(settings.logging.level)?;
 
+    let _file = lock_datadir(&settings.data.dir)?;
```

Comment:
> That `Settings` idea might not be so gooed because the settings and the file lock are actually separate concerns 🤔

Yeh wasn't completely sure about this. I think we should leave it main for now until there is an obvious and fitting place where to keep it.

Reply:
> If you would change some of the later function calls to use this file, returning it would not be useless :)
> 
> Otherwise, name it something like `_file_lock_to_ensure_we_have_exclusive_access_to_the"_config_until_this_is_dropped`.

The file lock ensures exclusive access to the datadir containing the seed.pem and the sqlite db file not the config.

Reply:
Well, the config also needs to be exclusive otherwise you will have conflict on the ports cnd binds to. But that will be solved as cnd will just crash because it cannot bind to said ports.